hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -587,4 +587,15 @@ public CertificateClient getCertificateClient() {
   public void setCertificateClient(CertificateClient client) {
     dnCertClient = client;
   }
+
+  @Override
+  public void printError(Throwable error) {
+    if (error.getMessage() == null || error.getMessage().length() == 0) {
+      //message could be null in case of NPE. This is unexpected so we can
+      //print out the stack trace.
+      LOG.error("Exception in HddsDatanodeService.", error);
+    } else {
+      LOG.error(error.getMessage().split("\n")[0]);
```

Comment:
Guess can keep this only for both the cases :
```
 LOG.error("Exception in HddsDatanodeService.", error);
```

Sometimes having trace of the exception really helps